core/src/main/java/org/apache/carbondata/core/datastorage/store/compression/nondecimal/UnCompressNonDecimalByte.java

```diff
@@ -85,12 +85,15 @@
   }
 
   @Override public CarbonReadDataHolder getValues(int decimal, Object maxValueObject) {
```

Comment:
This is not the way you supposed to do, please check how long data type is supported.